csep/utils/plots.py

```diff
@@ -553,7 +553,199 @@ def get_hist(x, mws, normed=True):
+def plot_basemap(basemap, extent, ax=None,  coastline=True, borders=False, linecolor='black', linewidth=True,
+                 grid=False, grid_labels=False, set_global=False, show=False):
+    """ Wrapper function for multiple cartopy base plots, including access to standard raster webservices
+
+     Args:
+         basemap (str): Possible values are: stock_img, stamen_terrain, stamen_terrain-background, google-satellite, ESRI_terrain, ESRI_imagery, ESRI_relief, ESRI_topo, ESRI_terrain, or webservice link (see examples in :func:`csep.utils.plots._get_basemap`. Default is None
+         extent (list):  [lon_min, lon_max, lat_min, lat_max]
+         show (bool): Flag if the figure is displayed
+         set_global (bool): Display the complete globe as basemap
+         coastline (str): Flag to plot coastline. default True,
+         borders (bool): Flag to plot country borders. default False,
+         linewidth (float): Line width of borders and coast lines. default 1.5,
+         linecolor (str): Color of borders and coast lines. default 'black',
+         grid (bool): Draws a grid in the basemap
+         grid_labels (bool): Annotate grid values
+
+     Returns:
+         :class:`matplotlib.pyplot.ax` object
+
+     """
+    if ax is None:
+        fig = pyplot.figure()
+        ax = fig.add_subplot(111, projection=ccrs.PlateCarree(central_longitude=0.0))
+    if set_global:
+        ax.set_global()
+
+    try:
+        # Set adaptive scaling
+        line_autoscaler = cartopy.feature.AdaptiveScaler('110m', (('50m', 50), ('10m', 5)))
+        tile_autoscaler = cartopy.feature.AdaptiveScaler(5, ((6, 50), (7, 15)))
+        tiles = None
+        # Set tile depth
+        tile_depth = 4 if set_global else tile_autoscaler.scale_from_extent(extent)
+        if coastline:
+            ax.coastlines(color=linecolor, linewidth=linewidth)
+        if borders:
+            borders =  cartopy.feature.NaturalEarthFeature('cultural', 'admin_0_boundary_lines_land',
+                                                           line_autoscaler, edgecolor=linecolor, facecolor='never')
+            ax.add_feature(borders, linewidth=linewidth)
+        if basemap == 'stock_img':
+            ax.stock_img()
+        elif basemap is not None:
+            tiles = _get_basemap(basemap)
+        if tiles:
+            ax.add_image(tiles, tile_depth)
+    except:
+        print("Unable to plot basemap. This might be due to no internet access, try pre-downloading the files.")
+
+    # Gridline options
+    if grid:
+        gl = ax.gridlines(draw_labels=grid_labels, alpha=0.5)
+        gl.right_labels = False
+        gl.xformatter = LONGITUDE_FORMATTER
+        gl.yformatter = LATITUDE_FORMATTER
+
+    if show:
+        pyplot.show()
+
+    return ax
+
+
+def plot_catalog(catalog, ax=None, show=False, extent=None, set_global=False, plot_args=None):
+    """ Plot catalog in a region
+
+    Args:
```

Comment:
we use a different format for the doc strings. you should be able to set up pycharm to use it. for the record, we are using google style docstrings. 

we can merge for now, because setting up linting and standardizing the code is an item on the to-do list.